samples/snippets/src/main/java/com/example/spanner/PgSpannerSample.java

```diff
@@ -0,0 +1,1603 @@
+                    + "  VenueId      BIGINT NOT NULL,"
+                    + "  Revenue      BIGINT, "
+                    + "  LastUpdateTime TIMESTAMPTZ NOT NULL,"
+                    + "  PRIMARY KEY (SingerId, VenueId))"
+                    + "  INTERLEAVE IN PARENT Singers ON DELETE CASCADE"),
+            null);
+    try {
+      // Initiate the request which returns an OperationFuture.
+      op.get();
+      System.out.println("Created Performances table in database: [" + id + "]");
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_create_table_with_timestamp_column]
+
+  // [START spanner_insert_data_with_timestamp_column]
+  static void writeExampleDataWithTimestamp(DatabaseClient dbClient) {
+    List<Mutation> mutations = new ArrayList<>();
+    for (SpannerSample.Performance performance : PERFORMANCES) {
+      mutations.add(
+          Mutation.newInsertBuilder("Performances")
+              .set("SingerId")
+              .to(performance.singerId)
+              .set("VenueId")
+              .to(performance.venueId)
+              .set("Revenue")
+              .to(performance.revenue)
+              .set("LastUpdateTime")
+              .to(Timestamp.now())
+              .build());
+    }
+    dbClient.write(mutations);
+  }
+  // [END spanner_insert_data_with_timestamp_column]
+
+  static void queryPerformancesTable(DatabaseClient dbClient) {
+    // Rows without an explicit value for Revenue will have a Revenue equal to
+    // null. A try-with-resource block is used to automatically release resources held by
+    // ResultSet.
+    try (ResultSet resultSet =
+             dbClient
+                 .singleUse()
+                 .executeQuery(
+                     Statement.of(
+                         "SELECT singerid as \"SingerId\", venueid as \"VenueId\", "
+                             + "revenue as \"Revenue\", lastupdatetime as \"LastUpdateTime\" "
+                             + "FROM Performances ORDER BY LastUpdateTime DESC"))) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %d %s %s\n",
+            resultSet.getLong("SingerId"),
+            resultSet.getLong("VenueId"),
+            // We check that the value is non null. ResultSet getters can only be used to retrieve
+            // non null values.
+            resultSet.isNull("Revenue") ? "NULL" : resultSet.getLong("Revenue"),
+            resultSet.getTimestamp("LastUpdateTime"));
+      }
+    }
+  }
+
+  // [START spanner_write_data_for_struct_queries]
+  static void writeStructExampleData(DatabaseClient dbClient) {
```

Comment:
Maybe I'm missing something here, but I don't see any structs here. I also don't think we support structs for PostgreSQL databases, so this sample is probably void.